modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/java/type/runtime/SourceTypeWrapper.java

```diff
@@ -99,9 +101,16 @@ public SourceTypeWrapper(Class<T> aClass, ClassTypeLoader typeLoader) {
     } else if (methods.size() > 1) {
       throw new IllegalSourceModelDefinitionException(
                                                       format("Source declared in class '%s' declares more than one method annotated with '%s'",
-                                                             aClass.getName(), annotationType.getSimpleName()));
+                                                             aClass.getName(), annotationTypes[0].getSimpleName()));
```

Comment:
Can you explain the  `[0]`? what if the repetition comes from another annotation?

Reply:
Yes, it is a private method that is intended only to be called within the class. The parameter is the list containing the same annotations, the legacy one and the sdk api one, and as they have the same name , that's why I used the first one.